### PR TITLE
Display terminal view on OAuth error

### PR DESCRIPTION
**Pull request by oleksandrpravosudko-okta**
## Description:
* wrap auth-js errors with specialized error types
* show error view for 'terminal' errors in v1 and v2
* reject top level promise on 'non-recoverable' errors
* introduce user-facing error message for out-of-sync system clock

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [x] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-516591](https://oktainc.atlassian.net/browse/OKTA-516591)

### Screenshot/Video:

https://user-images.githubusercontent.com/71440851/187463449-ea9fa5d0-242c-4934-badf-bec7bc2ac39e.mov

https://user-images.githubusercontent.com/71440851/187463737-f14b9cc7-8458-47fd-af08-ecd38341c417.mov

https://user-images.githubusercontent.com/71440851/187465870-4cd4afbf-51f2-4432-8154-888c5de88356.mov

### Downstream Monolith Build:

https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-87ecbc8-6319c176&page=1&pageSize=6&tab=main

### CODE REVIEW

File: src/util/OAuthErrors.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import { AuthSdkError, FieldError, OAuthError as SdkOAuthError} from '@okta/okta-auth-js';
+import * as Errors from './Errors';
```
Comment (aarongranick-okta) [inline src/util/OAuthErrors.ts:2]: @denysoblohin-okta has an approved PR which changes this file to expose named exports. Before merging this PR, I recommend waiting for PR https://github.com/okta/okta-signin-widget/pull/2763 to be merged, then update and resolve the conflict.

**Review by aarongranick-okta — APPROVED** [review]

**Comment by oleksandrpravosudko-okta** [comment]
@mauriciocastillosilva-okta, @nikhilvenkatraman-okta, @zihanwang-okta can you review please?

**Review by nikhilvenkatraman-okta — APPROVED** [review]
i18n changes look good.

**Review by zihanwang-okta — APPROVED** [review]
Changes in `i18n/*` lgtm.

**Pull request merged**